lib/config/fileconf.go

```diff
@@ -198,7 +200,14 @@ func MakeSampleFileConfig(flags SampleFlags) (fc *FileConfig, err error) {
 		g.DataDir = defaults.DataDir
 	}
 
-	g.AuthToken = flags.AuthToken
+	if flags.JoinMethod != "" && flags.JoinMethod != "token" {
+		g.JoinParams = JoinParams{
```

Comment:
Is `JoinParams{Method: "token", TokenName: "secret"}` valid for token joining? If so, maybe it would be simpler to always use join params.